server/scsynth/SC_PulseAudio.cpp

```diff
@@ -0,0 +1,319 @@
+        int numBufs = numSamples / bufFrames;
+
+        float* inBuses = mWorld->mAudioBus + mWorld->mNumOutputs * bufFrames;
+        float* outBuses = mWorld->mAudioBus;
+        int32* inTouched = mWorld->mAudioBusTouched + mWorld->mNumOutputs;
+        int32* outTouched = mWorld->mAudioBusTouched;
+
+        int bufFramePos = 0;
+        int64 oscTime = mOSCbuftime = (uint64)((m_DLL.PeriodTime() + m_maxOutputLatency) * kSecondsToOSCunits + .5);
+        int64 oscInc = mOSCincrement = (uint64)((m_DLL.Period() / numBufs) * kSecondsToOSCunits + .5);
+        mSmoothSampleRate = m_DLL.SampleRate();
+        double oscToSamples = mOSCtoSamples = mSmoothSampleRate * kOSCtoSecs /* 1/pow(2,32) */;
+
+        // main loop
+        for (int i = 0; i < numBufs; ++i, mWorld->mBufCounter++, bufFramePos += bufFrames) {
+            int32 bufCounter = mWorld->mBufCounter;
+            int32* tch;
+
+            // Process inputs, marking them as touched
+            tch = inTouched;
+            for (int k = 0; k < m_inputChannelCount; ++k) {
+                *tch++ = bufCounter;
+                float* dst = inBuses + k * bufFrames;
+                const float* src = (float*)inputBuffer + k + (bufFramePos * m_inputChannelCount);
+                for (int n = 0; n < bufFrames; n++) {
+                    *dst = *src;
+                    src += m_inputChannelCount;
+                    dst++;
+                }
+            }
+
+            // run engine
+            int64 schedTime;
+            int64 nextTime = oscTime + oscInc;
+            while ((schedTime = mScheduler.NextTime()) <= nextTime) {
+                float diffTime = (float)(schedTime - oscTime) * oscToSamples + 0.5;
+                float diffTimeFloor = floor(diffTime);
+                world->mSampleOffset = (int)diffTimeFloor;
+                world->mSubsampleOffset = diffTime - diffTimeFloor;
+
+                if (world->mSampleOffset < 0)
+                    world->mSampleOffset = 0;
+                else if (world->mSampleOffset >= world->mBufLength)
+                    world->mSampleOffset = world->mBufLength - 1;
+
+                SC_ScheduledEvent event = mScheduler.Remove();
+                event.Perform();
+            }
+            world->mSampleOffset = 0;
+            world->mSubsampleOffset = 0.0f;
+
+            World_Run(world);
+
+            // Process outputs (considering which ones have been touched)
+            tch = outTouched;
+            for (int k = 0; k < m_outputChannelCount; ++k) {
+                float* dst = (float*)outputBuffer + k + (bufFramePos * m_outputChannelCount);
+                if (tch[k] == bufCounter) {
+                    const float* src = outBuses + k * bufFrames;
+                    for (int n = 0; n < bufFrames; n++) {
+                        *dst = *src;
+                        src++;
+                        dst += m_outputChannelCount;
+                    }
+                } else {
+                    *dst = 0;
```

Comment:
Lines 180-181 need to read as follows. It's not sufficient to zero out only one sample here.

```
		    for (int n = 0; n < bufFrames; n++) {
                        *dst = 0;
                        dst += m_outputChannelCount;
		    }
```

Reply:
Thanks for the comments. From the symptoms, I was suspecting a bug in that area of code. I'll fix this tonight.

Reply:
@jamshark70 @dyfer , I just committed a fix for the issue when stopping the synths. Thanks a lot for raising, pinpointing and giving a solution for the issue. If you could give it a try and let me know if that fixes the problem for you, that would be great.

Reply:
Thanks for the update - I've built and tested this and I don't see this issue anymore.

Reply:
Ok, thanks for the update. I'll wait for @jamshark70 feedback before marking this as resolved,

Reply:
I had tested that change before proposing it, so I don't need to do it again.